### PR TITLE
[JENKINS-68785] No log messages for inbound agents after the first message (regression in 2.310)

### PR DESCRIPTION
**Pull request by basil**
Reverts the changes to `DefaultJnlpSlaveReceiver` from #5680 which were incorrect. The return value of `hudson.slaves.SlaveComputer#openLogFile` is not intended to be closed by any consumer; the close is handled by `hudson.slaves.SlaveComputer#kill`. If we close it then any further log statements after the initial "Inbound agent connected from…" will not be logged. This PR fixes the regression by restoring the original behavior from before #5680. I verified that when connecting an _inbound_ agent of an unsupported version that the log statements showed up properly after this change.

### Proposed changelog entries

Show all log messages when an inbound agent fails to connect (regression in 2.310).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6672"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by StefanSpieker — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by daniel-beck** [comment]
Is this related to the not great logging of https://www.jenkins.io/doc/upgrade-guide/2.346/#reject-connections-from-unsupported-agent-versions ?

**Comment by MarkEWaite** [comment]
Yes, this bug was discovered while investigating the logging of inbound agent disconnects.  Outbound agent disconnects log display the full message, while inbound agents do not until this change is implemented

**Comment by basil** [comment]
> inbound agents do not log the message until this change is implemented

Yes. In fact, inbound agents do not log _any_ messages (other than the first line) until this change is implemented.

**Pull request merged**
